src/stores/room-list-v3/RoomListStoreV3.ts

```diff
@@ -194,7 +194,7 @@ export class RoomListStoreV3Class extends AsyncStoreWithClient<EmptyObject> {
             case "MatrixActions.Room.myMembership": {
                 const oldMembership = getEffectiveMembership(payload.oldMembership);
                 const newMembership = getEffectiveMembershipTag(payload.room, payload.membership);
-                if (oldMembership === EffectiveMembership.Join && newMembership === EffectiveMembership.Leave) {
+                if (newMembership === EffectiveMembership.Leave) {
```

Comment:
I'm not sure this is right, if you get kicked from a room it should remain in your room list until it is forgotten otherwise you don't know you got kicked